game/world/managers/objects/units/creature/ThreatManager.py

```diff
@@ -14,11 +14,11 @@ class ThreatManager:
     def __init__(self,
                  owner: UnitManager):
         self.owner = owner
-        self.holders: Dict[int, ThreatHolder] = {}
+        self.holders: dict[int, ThreatHolder] = {}
         self.current_holder: Optional[ThreatHolder] = None
 
     def add_threat(self, source: UnitManager, threat: float):
-        if source != self:
+        if source != self and threat > 0.0:
```

Comment:
Adding negative threat (`mod_threat` instead?) should be possible for threat reduction effects.

Reply:
I thing I understood what u mean. Fixed.